lib/inc/internal/facts/linux/processor_resolver.hpp

```diff
@@ -14,12 +14,40 @@ namespace facter { namespace facts { namespace linux {
     struct processor_resolver : posix::processor_resolver
     {
      protected:
+        enum ArchitectureType {POWER, GENERIC};
```

Comment:
Perhaps use an `enum class`? I might also call this `X86` or something instead of `GENERIC`

Reply:
Nice! I didn't know about enum classes. I chose GENERIC to capture all architectures that aren't POWER8 -- you can have x86, and i386. Maybe NOTPOWER? I don't want to use a boolean for this either, because in the future we could have different architectures that could be added to this enum.

Reply:
Well, x86 as a [family](https://en.wikipedia.org/wiki/X86) would cover x86_64, i386, etc